W-14758017 AEH Jan 2024 release jh (#20)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/api-experience-hub-lightning-components.adoc

//* Show categories
//* Show tags

== Asset Version 

Shows the API version from Exchange when only one version is visible.

== Version State

Shows the API version lifecycle state.
* Asset Documentation Viewer

The API Documentation Viewer component shows an API’s documentation from Anypoint Exchange. The documentation can have multiple pages, and each page can have multiple sections.

== Assets Carousel

The API Carousel component shows a horizontally scrollable set of API cards that displays on the Home page. Each card shows an API.

You can specify values for the following Assets Carousel properties:

* Title
+
Title of the carousel.
* Subtitle
+
A subtitle that displays after the title of the carousel.

* Cards to display
+
The number of cards to display in the carousel. The carousel can display a minimum of three cards and a maximum of fifteen cards.

=== Edit the Titles in the API Carousel
You can edit the title and subtitle that appears for the API Carousel.

. From the Manage your API portal pages, click *Preview and publish your portal*.
. In the *Home* page, click the *Asset Carousel* area to edit the Assets Carousel component.
. In the *Title* field, enter a name.

> . In the *Subtitle* fieldI, enter a name.

Reviewer: When the user changes the number of cards, and then changes the carousel subtitle, this key {numberOfAssets} on the subtitle will replaced with the number of Cards to display the set:
Description added for this field: `Modify subtitle of carousel, '{numberOfAssets}' will be replaced with the numbers of assets displayed`

@JennyHajee Eval if this need to add this

Author: Thanks, I will add the steps to add the card count to the subtitle

= API Experience Hub Lightning Web Components 
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> You can design the pages of your API Experience Hub portal by adding and configuring Lightning web components in the Builder.

Reviewer 2: Just checking that Lightning web components is not a proper noun. If it is, then it probably should be init cap (LWC).

Reviewer 2: https://developer.salesforce.com/docs/platform/lwc/guide/get-started-release-notes.html

Reviewer 2: Global: Review the short descriptions to ensure you've covered the benefits.

Author: I double-checked. It should be Lightning components since Lightning Web components refers to the programming model.

These Lightning web components include both the Lightning web components available in all Salesforce Digital Experiences and the API Experience Hub Lightning web components available only in API Experience Hub.

Each API Experience Hub Lightning web component has different configuration options and adds different functionality to the portal.

Many API Experience Hub Lightning web components show data from Anypoint Platform. Portal admins can change some of these data fields, such as API names, descriptions, and icons. API Experience Hub saves these changes but does not push the changes back to Anypoint Platform.

> To add API Experience Hub components, click the *Components* lightning bolt icon to display the component palette. You can drag any component from the component palette to the current page. Select the component on the page to show the component’s configuration options. You can edit the values by entering or selecting values in the component property fields. 

Reviewer 2: use a screenshot of the icon here?

Author: I can add the icon post GA.

> To add API Experience Hub components, click the *Components* lightning bolt icon to display the component palette. You can drag any component from the component palette to the current page. Select the component on the page to show the component’s configuration options. You can edit the values by entering or selecting values in the component property fields. 

Reviewer 2: Global: Consider rewriting to remove "You can".

> To add API Experience Hub components, click the *Components* lightning bolt icon to display the component palette. You can drag any component from the component palette to the current page. Select the component on the page to show the component’s configuration options. You can edit the values by entering or selecting values in the component property fields. 

Reviewer 2: Would this be better as a numbered list (steps)?

Author: Yes, that will work.

> To delete a component, click the delete icon and click *Delete* from the dialog.

Reviewer 2: ```suggestion
To delete a component, click the delete icon and click *Delete* in the dialog.
```

+
The component shows the public API instances and documentation from Anypoint Exchange.

== Asset Detail Header

The Asset Details Header component displays the API header, including icon, name, contact email, and description.

The API icon, name, and description shown are set in the API Experience Hub control panel when you add an API version to your community. The contact email is set in Anypoint Exchange.

//You can enter or select the following Asset API Header properties:
//
//* Record id
//* Show API Contact
//* Show API Clients
//* Show instance URL
//* Show categories
//* Show tags

== Asset Version 

Shows the API version from Exchange when only one version is visible.

== Version State

Shows the API version lifecycle state.
* Asset Documentation Viewer

The API Documentation Viewer component shows an API’s documentation from Anypoint Exchange. The documentation can have multiple pages, and each page can have multiple sections.

> == Assets Carousel

Reviewer 2: Is there a way to rewrite this section to be less document-the-UI? Maybe just a screenshot with callouts? Do you need to describe the title and subtitle?

Author: I'd like to discuss this more with you after the release.

The API Carousel component shows a horizontally scrollable set of API cards that displays on the Home page. Each card shows an API.

You can specify values for the following Assets Carousel properties:

* Title
+
Title of the carousel.
* Subtitle
+
A subtitle that displays after the title of the carousel.

* Cards to display
+
The number of API cards to display in the carousel. The carousel can display a minimum of three cards and a maximum of fifteen cards. If the number of cards to display is five and the portal has only three APIs, then the carousel displays three cards instead of five.

> === Edit the Titles in the API Carousel

Reviewer 2: Is the user task "edit" or is it "change"?

Author: Seems like they have the same meaning. Which do you suggest?

> You can modify the title and subtitle that appear for the API Carousel. You can also show the number of cards in the subtitle by entering the `{numberOfAssets}` variable in the *Subtitle* field. For example, *Latest APIs added {numberOfAssets}*. The following image shows the subtitle as *Latest APIs added 5* since the variable displays the number selected in the *Cards to display* field.

Reviewer 2: for -> in?
since -> because

Author: got it

> image::aeh-builder-carousel-subtitle.png[Image of Asset Carousel Subtitle option,width=70%]

Reviewer 2: Do we hard-code image sizes?

. On the *Manage your API portal* page, click *Preview and publish your portal*.
. On the *Home* page, click the *Asset Carousel* area to edit the Assets Carousel component.

> . In the *Title* field, enter a name.

Reviewer 2: Are 77-79 necessary? Can you just say "Complete the fields" and include any necessary info?

. In the *Subtitle* field, enter a name.
. If you want to include the number of cards in the subtitle, enter `{numberOfAssets}` to the *Subtitle* field. 

> . Click the *Preview* button to confirm your changes.

Reviewer 2: Are you confirming your changes or previewing them?

. Click the *Publish* button to publish your changes.

> === Edit the Number of Cards to Display in the API Carousel

Reviewer 2: Edit -> Change?

The user's task is to change, which they accomplish by editing.

Reviewer 2: I think this is unnecessary, since you've got it covered above?

> You can edit the number of API cards that display in the API Carousel. 

Reviewer 2: You can -> To

> You can edit the number of API cards that display in the API Carousel. 

Reviewer 2: ```suggestion
To change the number of API cards that display in the API Carousel:
```

> . On the *Manage your API portal* page, click *Preview and publish your portal*.

Reviewer 2: Are these UI elements sentence case?

. On the *Home* page, click the *Asset Carousel* area to edit the Assets Carousel component.
. In the *Cards to display* field, select a number from three to fifteen from the list. 
. Click the *Preview* button to confirm your changes.
. Click the *Publish* button to publish your changes.

== Assets Catalog

The API Catalog component offers API consumers an interactive way to explore all the available APIs in an API portal. The API Catalog shows API cards in rows and columns, divided into pages, and shows controls to search the displayed APIs by categories and search terms. Each card shows an API. With the catalog, users can view and explore large numbers of API versions.

> == Search Box

Reviewer 2: What's the user task here?

Reviewer 2: Are these UI elements sentence case?

The API Experience Hub Search Box component enables users to search for both content that exists in Salesforce Experience Cloud (such as knowledge articles, support cases, and announcements) and content that exists in Anypoint Platform (such as APIs and client applications). Knowledge article, support cases, and announcements are currently not available.

> == Search Results

Reviewer 2: Should this be part of the previous section?

> The API Experience Hub Search Results component shows results after a user enters a search in the Search Box component.

Reviewer 2: Is this necessary to say? Or is it obvious?
